src/graphics/reproject.js

```diff
@@ -0,0 +1,84 @@
+import { Vec4 } from '../math/vec4.js';
+
+import {
+    TEXTURETYPE_RGBM, TEXTURETYPE_RGBE,
+    PIXELFORMAT_RGB16F, PIXELFORMAT_RGB32F, PIXELFORMAT_RGBA16F, PIXELFORMAT_RGBA32F
+} from './graphics.js';
+import { drawQuadWithShader } from './simple-post-effect.js';
+import { shaderChunks } from './chunks.js';
+import { RenderTarget } from './render-target.js';
+
+// get a coding string for texture based on its type and pixel format.
+var getCoding = function (texture) {
+    switch (texture.type) {
+        case TEXTURETYPE_RGBM:
+            return "RGBM";
+        case TEXTURETYPE_RGBE:
+            return "RGBE";
+        default:
+            switch (texture.format) {
+                case PIXELFORMAT_RGB16F:
+                case PIXELFORMAT_RGB32F:
+                case PIXELFORMAT_RGBA16F:
+                case PIXELFORMAT_RGBA32F:
+                    return "Linear";
+                default:
+                    return "Gamma";
+            }
+    }
+};
+
+/**
+ * @static
+ * @function
+ * @name pc.reproject
+ * @description Reproject a cubemap into equirectangular layout or visa-versa
```

Comment:
Why would you reproject from 6-face to equirectangular? What's the use case there? Also, does this function require source to be 2D texture and target to be a cubemap and vice versa?

Reply:
I imagine users may want to convert their cubemaps to equi when we add native support for equi in the engine (which most likely will be done at tool time). However supporting cubemap -> equi conversion when you have equi -> cubemap is practically free and there's no reason -not- to support it.

Reply:
Also, the function supports equi -> equi and cubemap -> cubemap (while calculating glossmaps and converting between pixel formats). So this function will ultimately replace https://github.com/playcanvas/engine/blob/master/src/graphics/prefilter-cubemap.js#L35 for the usual cubemap case.